File: src/main/java/habbits_tracker/habbits_tracker/task/dto/Task.java
package habbits_tracker.habbits_tracker.task.dto;

import habbits_tracker.habbits_tracker.task.utils.TaskPriority;
import habbits_tracker.habbits_tracker.task.utils.TaskStatus;
import jakarta.validation.constraints.*;

import java.time.LocalDate;

public record Task(

        @Null
        Long id,

        @NotBlank
        @Size(max = 120)
        String title,

        @Size(max = 1024)
        String description,

        @Null
        Long creatorId,

        @NotNull
        Long assignedUserId,

        TaskStatus status,

        @NotNull
        LocalDate createDateTime,

        @FutureOrPresent
        @NotNull
        LocalDate deadlineDate,

        TaskPriority priority
) {
}

File: frontend/src/types/task.ts
export type TaskStatus = 'CREATED' | 'IN_PROGRESS' | 'DONE';
export type TaskPriority = 'LOW' | 'MEDIUM' | 'HARD';

export interface Task {
  id: number | null;
  title: string;
  description: string | null;
  creatorId: number | null;
  assignedUserId: number;
  status: TaskStatus | null;
  createDateTime: string;
  deadlineDate: string;
  priority: TaskPriority | null;
}

